astropy/units/structured.py

```diff
@@ -515,3 +517,40 @@ def __ne__(self, other):
             other = other.item()
 
         return self.item() != other
+
+
+def _structured_unit_like_dtype(unit: UnitBase | StructuredUnit, dtype: np.dtype) -> StructuredUnit:
+    """Make a `StructuredUnit` of one unit, with the structure of a `numpy.dtype`.
+
+    Parameters
+    ----------
+    unit : UnitBase
+        The unit that will be filled into the structure.
+    dtype : `numpy.dtype`
+        The structure for the StructuredUnit.
+
+    Returns
+    -------
+    StructuredUnit
+    """
+    # Make sure dtype is structured
+    if dtype.names is None or dtype.fields is None:
+        raise ValueError(f"dtype {dtype} is not structured")
+    # if unit is structured, it should match the dtype
+    elif isinstance(unit, StructuredUnit):
```

Comment:
SImilarly here, why not just return and let `Quantity` raise an error later? For `Quantity` itself at least, one cannot even get here.

Reply:
removed.